connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala

```diff
@@ -938,11 +938,14 @@ class ClientE2ETestSuite extends RemoteSparkSession with SQLHelper with PrivateM
     }
     assert(e3.getMessage.contains("AMBIGUOUS_COLUMN_REFERENCE"))
 
-    val e4 = intercept[AnalysisException] {
-      // df1("i") is ambiguous as df1 appears in both join sides (df1_filter contains df1).
-      df1.join(df1_filter, df1("i") === 1).collect()
-    }
-    assert(e4.getMessage.contains("AMBIGUOUS_COLUMN_REFERENCE"))
+    //    val e4 = intercept[AnalysisException] {
+    //      // df1("i") is ambiguous as df1 appears in both join sides (df1_filter contains df1).
+    //      df1.join(df1_filter, df1("i") === 1).collect()
+    //    }
+    //    assert(e4.getMessage.contains("AMBIGUOUS_COLUMN_REFERENCE"))
+    //
+    //    "[AMBIGUOUS_COLUMN_OR_FIELD] Column or field `i` is ambiguous and has 2 matches.
```

Comment:
can not reproduce this failure with python client:
```
In [5]: df1 = spark.createDataFrame([(1, "a")], schema=["i", "j"])

In [6]: df1_filter = df1.filter(df1.i > 0)

In [7]: df2 = spark.createDataFrame([(2, "b")], schema=["i", "y"])

In [8]: df1.join(df1_filter, df1.i == 1).collect()
Out[8]: [Row(i=1, j='a', i=1, j='a')]
```

Reply:
ok, the reason it that spark connect scala client can not `collect` duplicated column names:

```
+---+---+---+---+
|  i|  j|  i|  j|
+---+---+---+---+
|  1|  a|  1|  a|
+---+---+---+---+
```

```
[info] - SPARK-45509: ambiguous column reference *** FAILED *** (2 seconds, 869 milliseconds)
[info]   org.apache.spark.sql.AnalysisException: [AMBIGUOUS_COLUMN_OR_FIELD] Column or field `i` is ambiguous and has 2 matches. SQLSTATE: 42702
[info]   at org.apache.spark.sql.errors.CompilationErrors.ambiguousColumnOrFieldError(CompilationErrors.scala:28)
[info]   at org.apache.spark.sql.errors.CompilationErrors.ambiguousColumnOrFieldError$(CompilationErrors.scala:23)
[info]   at org.apache.spark.sql.errors.CompilationErrors$.ambiguousColumnOrFieldError(CompilationErrors.scala:54)
[info]   at org.apache.spark.sql.connect.client.arrow.ArrowDeserializers$.$anonfun$createFieldLookup$1(ArrowDeserializer.scala:460)
[info]   at org.apache.spark.sql.connect.client.arrow.ArrowDeserializers$.$anonfun$createFieldLookup$1$adapted(ArrowDeserializer.scala:454)
[info]   at scala.collection.immutable.List.foreach(List.scala:334)
[info]   at org.apache.spark.sql.connect.client.arrow.ArrowDeserializers$.createFieldLookup(ArrowDeserializer.scala:454)
[info]   at org.apache.spark.sql.connect.client.arrow.ArrowDeserializers$.deserializerFor(ArrowDeserializer.scala:328)
[info]   at org.apache.spark.sql.connect.client.arrow.ArrowDeserializers$.deserializerFor(ArrowDeserializer.scala:86)
[info]   at org.apache.spark.sql.connect.client.arrow.ArrowDeserializingIterator.<init>(ArrowDeserializer.scala:542)
```